autocheese.lua

```diff
@@ -0,0 +1,155 @@
+--@module = true
+
+local ic = reqscript('idle-crafting')
+
+---make cheese using a specific barrel and workshop
+---@param barrel df.item
+---@param workshop df.building_workshopst
+---@return df.job
+function makeCheese(barrel, workshop)
+    ---@type df.job
+    local job = ic.make_job()
+    job.job_type = df.job_type.MakeCheese
+
+    local jitem = df.job_item:new()
+    jitem.quantity = 0
+    jitem.vector_id = df.job_item_vector_id.ANY_COOKABLE
+    jitem.flags1.unrotten = true
+    jitem.flags1.milk = true
+    job.job_items.elements:insert('#', jitem)
+
+    if not dfhack.job.attachJobItem(job, barrel, df.job_item_ref.T_role.Reagent, 0, -1) then
+        dfhack.error('could not attach item')
+    end
+
+    ic.assignToWorkshop(job, workshop)
+    return job
+end
+
+
+
+---unit is ready to take jobs
+---@param unit df.unit
+---@return boolean
+function unitIsAvailable(unit)
+    if unit.job.current_job then
+        return false
+    elseif #unit.individual_drills > 0 then
+        return false
+    elseif unit.flags1.caged or unit.flags1.chained then
+        return false
+    elseif unit.military.squad_id ~= -1 then
+        local squad = df.squad.find(unit.military.squad_id)
+        -- this lookup should never fail
+        ---@diagnostic disable-next-line: need-check-nil
+        return #squad.orders == 0 and squad.activity == -1
+    end
+    return true
+end
```

Comment:
shouldn't this function already be in ic? why not call that one to keep the logic unduplicated?

Reply:
The answer is that these functions do subtly different things:
1. The one for `idle-crafting` will not interrupt social activities
2. This one will also check for caged and chained units, which is handled elsewhere in `idle-crafting`. 

The good solution, after this is merged, is probably to have a function in the C++ `Units` module that will take a few flags to disable or enable optional checks.